### PR TITLE
add: deploy and enroll new chain

### PR DESCRIPTION
**Pull request by ErinHales**
- [ ] add test

### CODE REVIEW

File: typescript/optics-deploy/src/deployOptics.ts [hunk 1]
```diff
@@ -566,7 +566,7 @@ export async function deployNChains(chains: Deploy[]) {
 export async function deployAndEnrollNewChain(chains: Deploy[], newChain: Deploy) {
   await deployOptics(newChain);
 
-  chains.forEach((chain, i) => {
+  chains.forEach(async (chain, i) => {
```
Comment (prestwich) [inline typescript/optics-deploy/src/deployOptics.ts:569]: should this be

` await Promise.all(chains.map(async (chain, i) => { ... }));`

File: typescript/optics-deploy/src/core/index.ts [hunk 2]
```diff
@@ -563,6 +563,34 @@ export async function deployNChains(chains: CoreDeploy[]) {
...
+ *
+ * @param chains - An array of chain deploys
+ * @param newChain - A fresh deploy for the new chain
+ */
+export async function deployAndEnrollNewChain(chains: Deploy[], newChain: Deploy) {
+  await deployOptics(newChain);
+
+  await Promise.all(chains.map(async (chain, i) => {
+    await enrollRemote(chain, newChain);
+    await enrollRemote(newChain, chain);
```
Comment (prestwich) [inline typescript/optics-deploy/src/core/index.ts:580]: hmm, will this work if the previous deploy ownership has been transferred to governance?

Comment (anna-carroll) [reply typescript/optics-deploy/src/core/index.ts:580]: no, it wont :(

File: typescript/optics-deploy/src/core/index.ts [hunk 3]
```diff
@@ -563,6 +563,34 @@ export async function deployNChains(chains: CoreDeploy[]) {
...
+
+    if (i === 0) {
```
Comment (anna-carroll) [inline typescript/optics-deploy/src/core/index.ts:582]:
```
    if (i === 0) {
      await transferGovernorship(chain, newChain);
    }

    await relinquish(newChain);
```
this should be done outside of the `chains.map`. we are attempting to `relinquish` ownership of the `newChain` N times, where N = `chains.length` - this won't work

outside of the `map` statement,
```
      Deploy governorChain = chains[0];
      await transferGovernorship(governorChain, newChain);

      await relinquish(newChain);
```

File: typescript/optics-deploy/src/core/index.ts (hunk 2 above)
Comment (anna-carroll) [reply typescript/optics-deploy/src/core/index.ts:580]: in order to enroll the `newChain` in each existing `chain`, each existing chain will need the `governor` to send that transaction.

**Review by anna-carroll — CHANGES_REQUESTED** [review]
👯‍♀️

**Comment by prestwich** [comment]
closing this as requiring further consideration

**Pull request closed without merging**